### PR TITLE
[DOCS] EQL: Document `endsWith` function

### PR DESCRIPTION
**Pull request by jrodewig**
Adds documentation for the EQL `endsWith` function.

Relates to #54442.

### CODE REVIEW

File: docs/reference/eql/functions.asciidoc [hunk 1]
```diff
@@ -8,8 +8,71 @@ experimental::[]
...
+endsWith(<source>, <substring>)
+----
+
+*Parameters*
+
+`<source>`::
+(Required, string or `null`)
+Source string. If `null`, the function returns `null`.
++
+If using a field as the argument, this parameter only supports the
```
Comment (astefan) [inline docs/reference/eql/functions.asciidoc:61]: With this PR I realized one small aspect here. It should be valid for all other string functions, though.
One can use a `text` field in the function and there won't be any exception and the query will work **if** that `text` field will have a `keyword` sub-field.

For example, the following mapping
```
        "hostname" : {
            "type" : "text",
            "fields" : {
                "keyword" : {
                    "type" : "keyword",
                    "ignore_above" : 256
                }
            }
        }
```

and the query `process where endsWith(hostname, "foo") == true` won't trigger an error.

File: docs/reference/eql/functions.asciidoc [hunk 2]
```diff
@@ -8,8 +8,71 @@ experimental::[]
...
 {es} supports the following EQL functions:
 
+* <<eql-fn-endswith>>
 * <<eql-fn-substring>>
 
+[discrete]
+[[eql-fn-endswith]]
+=== `endsWith`
+
+Returns `true` if a source string ends with a provided substring.
```
Comment (astefan) [inline docs/reference/eql/functions.asciidoc:18]: This function is case insensitive, as well. And I believe it's worth being mentioned in docs.

File: docs/reference/eql/functions.asciidoc (hunk 1 above)
Comment (astefan) [reply docs/reference/eql/functions.asciidoc:61]: Valid statement for the other parameter, as well.

**Review by astefan — APPROVED** [review]
LGTM in general. Left one comment that should be valid for other string functions, as well. Thank you.

File: docs/reference/eql/functions.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/eql/functions.asciidoc]: Thanks! Addressed with 508be061d6be82a802863688765c46323f12283a.

File: docs/reference/eql/functions.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/eql/functions.asciidoc]: Thanks again for pointing this out. Addressed with 508be061d6be82a802863688765c46323f12283a.

**Comment by jrodewig** [comment]
### Backport commits
master e86e148 
7.x b43eb5a

**Pull request merged**